### PR TITLE
Scrum 37 shopping cart

### PR DESCRIPTION
**Pull request by BeattLau**
Scrum 55 - Create cart model
Scrum 56- Implement endpoints

### CODE REVIEW

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Autowired
+    CartService cartService;
+
+    @GetMapping("/cart/user/{userId}")
+    public ResponseEntity<List<Products>> getCartContents(@PathVariable Long userId) {
+        try {
+            List<Products> cartContents = cartService.getCartContents(userId);
+            return ResponseEntity.ok(cartContents);
+        } catch (UserNotFoundException e) {
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java]: Do you have a global exception handler? If yes then throw the exception form your service, if not you can refactor this later

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1")
+public class CartController {
+
+    @Autowired
+    CartService cartService;
+
+    @GetMapping("/cart/user/{userId}")
+    public ResponseEntity<List<Products>> getCartContents(@PathVariable Long userId) {
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java]: Don't you already know who the user is from his Auth details?

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            List<Products> cartContents = cartService.getCartContents(userId);
+            return ResponseEntity.ok(cartContents);
+        } catch (UserNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(Collections.emptyList());
+        }
+    }
+    @PostMapping("/cart/")
+    public ResponseEntity<List<Products>> addProductToCart(
+            @RequestBody Products products,
+            @RequestParam Long userId,
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java]: same

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            @RequestParam Long productId) {
+        try {
+            List<Products> addedProducts = cartService.addProductToCart(products, userId, productId);
+            return ResponseEntity.ok(addedProducts);
+        } catch (EntityNotFoundException e) {
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java:34]: same

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(Collections.emptyList());
+        }
+    }
+    @DeleteMapping("/cart/delete")
+    public ResponseEntity<String> deleteProductFromCart(
+            @RequestParam Long userId,
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java]: same

File: src/main/java/com/ecommerce/Controller/CartController.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            @RequestParam Long productId) {
+        try {
+            cartService.deleteProductFromCart(userId, productId);
+            return ResponseEntity.ok("Product removed from the cart successfully");
+        } catch (ProductNotFoundException e) {
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Controller/CartController.java:49]: same

File: src/main/java/com/ecommerce/Entity/Products.java [hunk 7]
```diff
@@ -5,15 +5,21 @@
...
 import lombok.Setter;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Entity
 @Table(name = "Products")
 @RequiredArgsConstructor @Getter @Setter
-public class Products{
+public class Products extends ShoppingCart {
```
Comment (faruka-startsteps) [inline src/main/java/com/ecommerce/Entity/Products.java]: why do your Products extend the shopping cart?  I don't see a need for this

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/com/ecommerce/Controller/CartController.java (hunk 2 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/CartController.java]: Updated, please see commit 681fcd0

File: src/main/java/com/ecommerce/Controller/CartController.java (hunk 3 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/CartController.java]: Updated, please see commit 681fcd0

File: src/main/java/com/ecommerce/Controller/CartController.java (hunk 5 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/CartController.java]: Updated, please see commit 681fcd0

File: src/main/java/com/ecommerce/Entity/Products.java (hunk 7 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Entity/Products.java]: Updated, please see commit 681fcd0

File: src/main/java/com/ecommerce/Controller/CartController.java (hunk 1 above)
Comment (BeattLau) [reply src/main/java/com/ecommerce/Controller/CartController.java]: I will refactor this later as I still have work to do on my global exception handler

**Review by f4r33 — APPROVED** [review]

**Pull request merged**